testing/tests/DevExpress.ui.widgets.scheduler/integration.appointmentsVertical.tests.js

```diff
@@ -1610,4 +1610,250 @@ QUnit.module('Integration: Appointments on vertical views (day, week, workWeek)'
+            });
+
+            QUnit.test(`Appointments should filter correctly if resources and scrolling.mode = "virtual", groupOrientation: ${groupOrientation}`, function(assert) {
+                const data = [
+                    {
+                        startDate: new Date(2016, 9, 5, 0, 0),
+                        endDate: new Date(2016, 9, 5, 0, 30),
+                        resourceId0: 0,
+                        text: 'test_00'
+                    },
+                    {
+                        startDate: new Date(2016, 9, 5, 2),
+                        endDate: new Date(2016, 9, 5, 3),
+                        resourceId0: 1,
+                        text: 'test_01'
+                    },
+                    {
+                        startDate: new Date(2016, 9, 5, 4),
+                        endDate: new Date(2016, 9, 5, 4, 1),
+                        resourceId0: 0,
+                        text: 'test_02'
+                    },
+                    {
+                        startDate: new Date(2016, 9, 5, 12),
+                        endDate: new Date(2016, 9, 5, 13),
+                        resourceId0: 1,
+                        text: 'test_03'
+                    },
+                    {
+                        startDate: new Date(2016, 9, 5, 18),
+                        endDate: new Date(2016, 9, 5, 20),
+                        resourceId0: 0,
+                        text: 'test_04'
+                    }
+                ];
+                this.createInstance({
+                    dataSource: data,
+                    currentDate: new Date(2016, 9, 5),
+                    views: [{
+                        type: 'day',
+                        groupOrientation: groupOrientation,
+                    }],
+                    currentView: 'day',
+                    scrolling: {
+                        mode: 'virtual'
+                    },
+                    resources: [{
+                        fieldExpr: 'resourceId0',
+                        dataSource: [
+                            { text: 'Rc0_0', id: 0, color: '#727bd2' },
+                            { text: 'Rc0_1', id: 1, color: '#32c9ed' },
+                        ],
+                        label: 'Resource0'
+                    }],
+                    height: 400
+                });
+
+                const filteredItems = this.instance.getFilteredItems();
+
+                assert.equal(filteredItems.length, 3, 'Filtered items length is correct');
+                assert.deepEqual(filteredItems[0], data[0], 'Filtered item 0 is correct');
+                assert.deepEqual(filteredItems[1], data[1], 'Filtered item 1 is correct');
+                assert.deepEqual(filteredItems[2], data[2], 'Filtered item 2 is correct');
+            });
+        });
+
+        QUnit.test('Appointments should filter correctly if groups, resources and scrolling.mode = "virtual", groupOrientation: "vertical"', function(assert) {
```

Comment:
'Appointments should be filterer correctly when they are grouped vertically and scrolling.mode = "virtual"'